### PR TITLE
Add integrations, further cleanup

### PR DESCRIPTION
**Pull request by devkral**
Changes:
- cleanup more typings
- make types abstract and let base merge them with BaseModel
- make Scheduler a plain class (we cannot serialize it anyway, there is no benefit)
- enhance Scheduler:
  - add asgi integration
  - add contextmanager support
- update docs
- add store encryption

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -157,6 +157,60 @@ batteries that are currently used by the framework and leveraging Asyncz.
...
+=== "Starlette"
+
+    ```python
+    from asyncz.schedulers import AsyncIOScheduler
+
+    async lifespan(app):
+        with AsyncIOScheduler() as scheduler:
+            yield
+            # or yield a state
+    app = Starlette(
```
Comment (tarsil) [inline docs/index.md:207]: Replace Starlette with `Lilya` :D or if you want, add both but if you want just one, Lilya please.

Comment (devkral) [reply docs/index.md:222]: addes, but I have to test it

**Comment by devkral** [comment]
big missing point: tests

we need to test the integrations

**Comment by devkral** [comment]
but out of time for today

**Comment by tarsil** [comment]
Great stuff you have done here btw @devkral

**Review by tarsil — APPROVED** [review]

**Review by tarsil — APPROVED** [review]

**Pull request merged**